# Пришел новый датасет. Его надо изучить.

Скачиваем датасет. Я скачал его на компьютер и добавил через загрузку файла в папку с датасетами. Пробуем присвоить переменную и открыть один из датасетов.


In [49]:
# Загрузка модуля pandas
import pandas as pd

In [2]:
# Загрузка файла из облака в colab
import gdown
gdown.download('https://storage.yandexcloud.net/terratraineeship/23_InspectrumClinic_RS/datasets/26.06%20new%20dataset.zip', None, quiet=True)

'26.06%20new%20dataset.zip'

Распакуем зипархив.

In [3]:
import zipfile

# Указываем путь к скачанному зип-архиву
zip_path = '/content/26.06%20new%20dataset.zip'

# Указываем путь для распаковки архива
output_path = '/content/sample_data/'

# Открываем зип-архив
with zipfile.ZipFile(zip_path, 'r') as zip_ref:
    # Распаковываем все файлы из архива в указанную папку
    zip_ref.extractall(output_path)

print("Архив успешно распакован.")


Архив успешно распакован.


In [50]:
# Присвоение переменной
data = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_01-22.csv')


ParserError: ignored

Появилась ошибка Error tokenizing data. C error: Expected 1 fields in line 73, saw 5 , повествующая нам, что при попытке разделить данные в файле CSV на поля, произошла ошибка. Конкретно в строке 73 файла было обнаружено больше полей, чем ожидалось.
Значит попробуем открыть блокнотом на компудакторе. Посмотрим что там не так.

**Ошибка возникла, потому что строка содержит символы, которые могут быть распознаны как разделители полей, хотя они на самом деле являются частью данных. В данном случае, это символы точки с запятой (;).** Чтобы правильно обработать эту строку, следует указать правильные параметры при чтении файла CSV. Воспользуйтесь параметром delimiter (разделитель) и quotechar (символ кавычек), чтобы указать правильные символы разделителя и экранирования.

In [6]:

data = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_01-22.csv', delimiter=';', quotechar='"')


In [7]:
# Убираем ограничение на вывод колонок с 20 до 400import pandas as pd

pd.set_option('display.max_columns', 400)


In [ ]:
# Выводим пекрвые 5 строк для анализа.
data

Сработал метод. Открылся файл. Сразу вижу, что тут 1511 строк и 342 колонки, что несколько отличается от первого датасета в котором было 4500 колонок. а значит обрабатывать таблицу придется снова. Нужно будет удалить лишнее и разобраться почему таблицы разные и еще желательно сопоставить все таблицы датасета.**Отсутствуют названия колонок!** Датасетов 17 штук. Датасеты проименованы месяцем и годом, когда проходил медосмотр. Датасеты свободно открываются EXCEL, блокнотом. В датасете **neiro_06-23** есть названия колонок, в остальных нет, могу предположить, что названия совпадают со всеми данными в остальных датасетах, но нужно проверить.

**Есть более простой способ открытия и анализа датасетов. Заключается в пропуске проблемных строк. С помощью него присвоим каждому датасету новую переменную.**

In [46]:
data_01_22 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_01-22.csv', sep = ';', on_bad_lines='skip', engine='python' )
data_02_22 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_02-22.csv', sep = ';', on_bad_lines='skip', engine='python' )
data_03_22 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_03-22.csv', sep = ';', on_bad_lines='skip', engine='python' )
data_04_22 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_04-22.csv', sep = ';', on_bad_lines='skip', engine='python' )
data_05_22 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_05-22.csv', sep = ';', on_bad_lines='skip', engine='python' )
data_06_22 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_06-22.csv', sep = ';', on_bad_lines='skip', engine='python' )
data_07_22 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_07-22.csv', sep = ';', on_bad_lines='skip', engine='python' )
data_08_22 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro-08-22.csv', sep = ';', on_bad_lines='skip', engine='python' )
data_09_22 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro-09-22 (1).csv', sep = ';', on_bad_lines='skip', engine='python' )
data_10_22 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro-10-22.csv', sep = ';', on_bad_lines='skip', engine='python' )
data_12_22 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_12-22.csv', sep = ';', on_bad_lines='skip', engine='python' )
data_01_23 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_01_23.csv', sep = ';', on_bad_lines='skip', engine='python' )
data_02_23 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_02_23.csv', sep = ';', on_bad_lines='skip', engine='python' )
data_03_23 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_03_23.csv', sep = ';', on_bad_lines='skip', engine='python' )
data_04_23 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_04_23.csv', sep = ';', on_bad_lines='skip', engine='python' )
data_05_23 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_05-22.csv', sep = ';', on_bad_lines='skip', engine='python' )
data_06_23 = pd.read_csv('/content/sample_data/26.06 new dataset/neiro_06-23.csv', sep = ';', on_bad_lines='skip', engine='python' )

**Названия датасетов**
data_01_22,
data_02_22,
data_03_22,
data_04_22,
data_05_22,
data_06_22,
data_07_22,
data_08_22,
data_09_22,
data_10_22,
data_12_22,
data_01_23,
data_02_23,
data_03_23,
data_04_23,
data_05_23,
data_06_23,

In [48]:

datasets = [data_01_22, data_02_22, data_03_22, data_04_22, data_05_22, data_06_22,
            data_07_22, data_08_22, data_09_22, data_10_22, data_12_22, data_01_23,
            data_02_23, data_03_23, data_04_23, data_05_23, data_06_23]

for i, dataset in enumerate(datasets, start=1):
    num_columns = dataset.shape[1]
    print(f"Датасет {i}: Количество колонок: {num_columns}")



Датасет 1: Количество колонок: 342
Датасет 2: Количество колонок: 294
Датасет 3: Количество колонок: 446
Датасет 4: Количество колонок: 306
Датасет 5: Количество колонок: 262
Датасет 6: Количество колонок: 458
Датасет 7: Количество колонок: 218
Датасет 8: Количество колонок: 304
Датасет 9: Количество колонок: 310
Датасет 10: Количество колонок: 258
Датасет 11: Количество колонок: 222
Датасет 12: Количество колонок: 228
Датасет 13: Количество колонок: 290
Датасет 14: Количество колонок: 430
Датасет 15: Количество колонок: 266
Датасет 16: Количество колонок: 262
Датасет 17: Количество колонок: 190


Везде количество колонок разное. В 17 датасете data_06_23 есть названия колонок и их количество 190. Имеет смысл проанализировать его сначала, а потом склить с остальными датасетами, предварительно назвав колонки как надо.